checkmedia.c

```diff
@@ -1,527 +1,182 @@
+    media = mediacheck_init(device, progress);
   }
   else {
     update_device_list(0);
-    if(config.device) get_info(device = config.device);
+    if(config.device) {
+      device = config.device;
+      media = mediacheck_init(device, progress);
+    }
 
     dia_message("No CD-ROM or DVD found!!!", MSGTYPE_ERROR);
   }
 
-#if 0
-  if(iso.err) {
-    for(hd = fix_device_names(hd_list(config.hd_data, hw_cdrom, 0, NULL)); hd; hd = hd->next) {
-      if(hd->is.notready) continue;
-      get_info(device = hd->unix_dev_name);
-      if(!iso.err) break;
-    }
-  }
-
-  if(iso.err) {
-    if(dia_message("Insert Installation CD-ROM or DVD.", MSGTYPE_INFO)) return;
-
-    for(hd = fix_device_names(hd_list(config.hd_data, hw_cdrom, 0, NULL)); hd; hd = hd->next) {
-      if(hd->is.notready) continue;
-      get_info(device = hd->unix_dev_name);
-      if(!iso.err) break;
+  if(media) {
+    for(i = 0; i < sizeof media->tags / sizeof *media->tags; i++) {
+      if(!media->tags[i].key) break;
+      log_info("media tags: key = \"%s\", value = \"%s\"\n", media->tags[i].key, media->tags[i].value);
     }
   }
 
-  if(iso.err) {
-    dia_message("No CD-ROM or DVD found.", MSGTYPE_ERROR);
+  if(
+    !media ||
+    media->err ||
+    (media->iso_blocks && media->pad_blocks >= media->iso_blocks)
+  ) {
+    dia_message("This is not a SUSE medium.", MSGTYPE_ERROR);
     config.manual=1;
     return;
   }
-#endif
-
-  log_info(
-    "  app: %s\nmedia: %s%d\n size: %u kB\n  pad: %u kB\n",
-    iso.app_id,
-    iso.media_type,
-    iso.media_nr ?: 1,
-    iso.size,
-    iso.pad
-  );
-
-  log_info("  ref: ");
-  if(iso.digest.got_old) for(i = 0; i < iso.digest.size; i++) log_info("%02x", iso.digest.old[i]);
-  log_info("\n");
-
-  if(!*iso.app_id || !iso.digest.got_old || iso.pad >= iso.size) {
-    sprintf(buf, "This is not a %s medium.", config.product);
-    dia_message(buf, MSGTYPE_ERROR);
+
+  if(
+    !mediacheck_digest_valid(media->digest.iso) &&
+    !mediacheck_digest_valid(media->digest.part)
+  ) {
+    dia_message("No digests to check.", MSGTYPE_ERROR);
     config.manual=1;
     return;
   }
 
-  do_digest(device);
-
-  if(iso.err_ofs) {
-    log_info("  err: sector %u\n", iso.err_ofs >> 1);
-  }
+  if(media->app_id) log_info("app: %s\n", media->app_id);
 
-  log_info("check: ");
-  if(iso.digest.got_old) {
-    if(iso.digest.ok) {
-      log_info("%ssum ok\n", iso.digest.name);
-    }
-    else {
-      log_info("%ssum wrong\n", iso.digest.name);
-    }
-  }
-  else {
-    log_info("%ssum not checked\n", iso.digest.name);
+  if(media->iso_blocks) {
+    log_info(
+      "iso size: %u%s kB\n",
+      media->iso_blocks >> 1,
+      (media->iso_blocks & 1) ? ".5" : ""
+    );
   }
 
-  if(iso.digest.got_current) {
-    log_info("  %s: ", iso.digest.name);
-    for(i = 0; i < iso.digest.size; i++) log_info("%02x", iso.digest.full[i]);
-    log_info("\n");
-  }
+  if(media->pad_blocks) log_info("pad: %u kB\n", media->pad_blocks >> 1);
```

Comment:
above you add .5 here not. Is it imporant info for padding? BTW as majority of sizes are in 512 bytes, maybe to return size in Kb here can be method? like `if(media->pad_blocks) log_info("pad: %u kB\n", size_in_kb(media->pad_blocks));`

Reply:
padding is always in 2k units, so no odd block numbers possible